src/mlpack/methods/ann/layer/randomized_relu.hpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file methods/ann/layer/randomized_relu.hpp
+ * @author Shubham Agrawal
+ *
+ * @code
+ * @article{Bing Xu2015,
+ *  author = {Bing Xu, Naiyan Wang, Tianqi Chen, Mu Li},
+ *  title = {Empirical Evaluation of Rectified Activations in Convolutional 
+ *      Network},
+ *  year = {2015},
+ *  url = {https://arxiv.org/pdf/1505.00853.pdf}
+ * }
+ * @endcode
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_ANN_LAYER_RRELU_HPP
+#define MLPACK_METHODS_ANN_LAYER_RRELU_HPP
+
+#include <mlpack/prereqs.hpp>
+#include <mlpack/core/math/random.hpp>
+
+namespace mlpack {
+namespace ann /** Artificial Neural Network. */ {
+
+/**
+ * The RReLU activation function (alpha is randomized), defined by
+ *
+ * @f{eqnarray*}{
+ * f(x) &=& \max(x, alpha*x) \\
+ * f'(x) &=& \left\{
+ *   \begin{array}{lr}
+ *     1 & : x > 0 \\
+ *     alpha & : x \le 0
+ *   \end{array}
+ * \right.
+ * @f}
+ *
+ * @tparam InputDataType Type of the input data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ * @tparam OutputDataType Type of the output data (arma::colvec, arma::mat,
+ *         arma::sp_mat or arma::cube).
+ */
+template <
+    typename InputDataType = arma::mat,
+    typename OutputDataType = arma::mat
+>
+class RReLU
+{
+ public:
+  /**
+   * Create the RReLU object using the specified parameters.
+   * The non zero gradient will be adjusted by specifying the parameter
+   * lowerbound, and higherbound from range 1 to inf.
+   * Default (lowerbound = 3.0, higherbound = 8.0)
+   *
+   * @param lowerbound Lowerbound of uniform distribution
+   * @param higherbound Higherbound of uniform distribution
```

Comment:
```suggestion
   * @param lowerBound Lower bound of uniform distribution.
   * @param upperBound Upper bound of uniform distribution.
```
To be consistent with the rest of the codebase, can you adjust swap `lowerbound` with lowerBound` and `higherbound` with `upperBound` in the rest of the implementation as well?

Reply:
Done